src/screens/Member/signup.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import {
+    View,
+    Text,
+    TextInput,TouchableOpacity, Platform, KeyboardAvoidingView, ScrollView
+} from "react-native";
+import styles from "./styles";
+import Icon from 'react-native-vector-icons/FontAwesome';
+import {NavigationScreenProp} from "react-navigation";
+
+interface IProps {
+    navigation: NavigationScreenProp<any, any>
+}
+
+export default class Signup extends React.PureComponent<IProps, any> {
+
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            emailOrPhone: "",
+            password: "",
+            username: "",
+            name: "",
+            validate: null
+        }
+    }
+
+    // @ts-ignore
+    componentDidUpdate() {
+        this.validate();
+    }
+
+    validate = () => {
+        const {emailOrPhone, password, username, name} = this.state;
+
+        if (emailOrPhone.length > 5 && password.length > 5 && username && name) {
+            this.setState({
+                validate: true
+            })
+        } else {
+            this.setState({
+                validate: false
+            })
+        }
+    }
```

Comment:
Can also be simplified with a single line expression.
`this.setState({validate: emailOrPhone.length > 5 && password.length > 5})`